app/routers/update_router.py

```diff
@@ -23,19 +24,28 @@ class UpdateQueryParams(BaseModel):
 @router.post("", response_model=None, summary="Update all suppliers")
 def update_supplier(
     supplier_id: str | None = None,
+    override_ignore: Annotated[list[str] | None, Query()] = None,
     query_params: UpdateQueryParams = Depends(),
     service: UpdateConsumerService = Depends(get_update_consumer_service),
     supplier_provider: SupplierProvider = Depends(get_supplier_provider),
     supplier_ignored_directory_service: SupplierIgnoredDirectoryService = Depends(get_supplier_ignored_directory_service),
 ) -> Any:
     since = query_params.since.astimezone(timezone.utc) if query_params.since else None
```

Comment:
Currently the method consists of a flow for supplier_id is none and if not none. I think that we should break up the method for a single and all suppliers in two separate methods.